cmake/tpl/FindKLU.cmake

```diff
@@ -30,6 +30,19 @@
 #   KLU_LIBRARIES   - all of the libraries needed for KLU
 # ---------------------------------------------------------------
 
+if (NOT (KLU_INCLUDE_DIR OR KLU_LIBRARY_DIR OR KLU_LIBRARY))
+  # Prefer the import target from upstream SuiteSparse if it is available
+  # and the user didn't point to a specific (different) version.
+  find_package(KLU CONFIG)
```

Comment:
I haven't tried yet, but is it right? I am wondering if should it be `find_package(SuiteSparse CONFIG COMPONENTS KLU)`?

Reply:
SuiteSparse doesn't install one big Config file which contains information for the libraries in the form of components. Instead, the goal was to allow building each of the libraries as stand-alone projects.
To achieve that goal, SuiteSparse installs separate Config files for each of the libraries.

A similar change is used by MSYS2 to build SUNDIALS with newer versions of SuiteSparse:
https://github.com/msys2/MINGW-packages/commit/1dfb64a6ef97cfe5dc295bdf038a1d730b687081

So, that command is working.